bundle/Helper/ImportUrlsHelper.php

```diff
@@ -0,0 +1,250 @@
+<?php
+
+namespace Novactive\Bundle\eZSEOBundle\Helper;
+
+use Doctrine\ORM\EntityManagerInterface;
+use eZ\Publish\Core\SignalSlot\URLWildcardService;
+use Novactive\Bundle\eZSEOBundle\Entity\RedirectImportHistory;
+use Psr\Log\LoggerInterface;
+use Psr\Log\LogLevel;
+use Symfony\Component\Translation\TranslatorInterface;
+use Symfony\Component\Filesystem\Filesystem;
+
+
+/**
+ * Helper for import url.
+ */
+class ImportUrlsHelper
+{
+    /**
+     * @var URLWildcardService
+     */
+    private $urlWildCardService;
+
+    /**
+     * @var LoggerInterface
+     */
+    private $logger;
+
+    /**
+     * @var TranslatorInterface
+     */
+    private $translator;
+
+    /**
+     * @var EntityManagerInterface
+     */
+    private $entityManager;
+
+    /**
+     * @var Filesystem
+     */
+    private $fs;
+
+    private $webDirectory;
+
+    /**
+     * ImportUrlsHelper constructor.
+     * @param URLWildcardService $urlWildcardService
+     * @param EntityManagerInterface $entityManager
+     * @param $webDirectory
+     * @param TranslatorInterface $translator
+     * @param LoggerInterface $logger
+     * @param Filesystem $fileSystem
+     */
+    public function __construct(
+        URLWildcardService $urlWildcardService,
+        EntityManagerInterface $entityManager,
+        $webDirectory,
+        TranslatorInterface $translator,
+        LoggerInterface $logger,
+        Filesystem $fileSystem
+    ) {
+        $this->urlWildCardService = $urlWildcardService;
+        $this->entityManager = $entityManager;
+        $this->webDirectory = $webDirectory;
+        $this->translator = $translator;
+        $this->logger = $logger;
+        $this->fs = $fileSystem;
+
+    }
+
+    /**
+     * @param $filePath
+     * @return array
+     */
+    public function importUrlRedirection($filePath)
+    {
+        $counter = 0;
+        $params = $return = [];
+
+        if (($fileToImport = fopen($filePath, 'r')) !== false) {
+            $totalImported = 0;
+            $totalUrls     = 0;
+            //create file log
+            $this->fs->dumpFile(
```

Comment:
are we putting the log in the Web directory? 
please fix

Reply:
it is a file that represents the recap of the import of a csv file and I put it in the web directory because the administrator can download it later via the page Import Urls History

Reply:
yes but if that is exposed then it is public. That is not possible to do that.
Store in the cluster, and then create a controller to make the download.

Few pieces of code to store it in the cluster:

```yaml
services:
    # IO Service
    novaezmailing.storage.report.io_service:
        parent: ezpublish.core.io.service
        calls:
            - [ setPrefix, [ "novaezmailingreport" ] ]
```

then automatically you can inject and use that service:
```php
$ioService = $this->getIOService();
        // we have a file, assign it
        if ($file instanceof File) {
            // remove the previous
            if (null !== $entity->getFilePath()) {
                $ioFile = $ioService->loadBinaryFile($entity->getFilePath());
                $ioService->deleteBinaryFile($ioFile);
            }
            $uploadedFileStruct     = $ioService->newBinaryCreateStructFromLocalFile($file->getRealPath());
            $uploadedFileStruct->id = "a unique name for the file in the cluster";
            $ioFile                 = $ioService->createBinaryFile($uploadedFileStruct);
        }
```

And then the controller action should look like

```php
 $file     = $ioService->loadBinaryFile("a unique name for the file in the cluster");
$response = new Response($ioService->getFileContents($file));
 $disposition = $response->headers->makeDisposition(
            ResponseHeaderBag::DISPOSITION_INLINE,
            "{$clientFileName}.{$item->getFileExtension()}"
        );
        $response->headers->set('Content-Disposition', $disposition);
        $response->headers->set('Cache-Control', 'private');
        $response->headers->set('Content-type', 'application/octet-stream');

        return $response;
```

thanks